### PR TITLE
Ensure we disconnect any projection buffers we create for views

### PR DESCRIPTION
**Pull request by jasonmalinowski**
When we create a projection buffer, that buffer is strongly held by the underlying buffer, which means we have a memory leak unless we disconnect it by unmapping all spans. This also causes problems because bugs in the editor around projections can then cause crashes when we didn't actually need the buffers anymore.

**Customer scenario:** user mouses over a closing brace, and we show quick info help. At this point, we have a buffer leaked which due to editor bugs may later cause a crash.
**Bugs this fixes:** https://devdiv.visualstudio.com/DevDiv/_workitems/edit/378999
**Workarounds, if any:** don't mouse over closing braces which causes us to show tips.
**Risk:** low. We already use this pattern in other places; this one place was missed.
**Performance impact:** improved. Fixes a small memory leak. and over time enough of them might actually hurt typing performance.
**Is this a regression from a previous update?** No. The underlying editor bug has been there a long time; we've just finally began to understand the causes.
**Root cause analysis:** When you mouse over a closing brace, we create a "projection buffer" atop the main buffer that shows just the small portion of the file from the matching brace. It's not obvious in the editor API that unless you take particular steps to clean this up, your buffer will continue to live around later. Even the editor team has discovered another version of this mistake, and long term is considering changes to fix it. Not cleaning it up means it's leaked, and which then exposes other bugs where certain edits when there's a projection buffer cause the editor to crash. The editor is aware of those bugs too but is hesitant to change them given the extreme sensitivity of that code.
**How was the bug found?** Customer reports through Send Feedback, which motivated us to add additional telemetry for 15.3 Preview 3.

### CODE REVIEW

File: src/EditorFeatures/Core/Shared/Utilities/ViewHostingControl.cs [hunk 1]
```diff
@@ -45,26 +61,33 @@ private void OnIsVisibleChanged(object sender, DependencyPropertyChangedEventArg
...
             }
             else
             {
-                ((ITextView)this.Content).Close();
                 this.Content = null;
+
+                _createdView.Close();
+                _createdView = null;
+
+                if (_createdTextBuffer is IProjectionBuffer projectionBuffer)
```
Comment (sharwell) [inline src/EditorFeatures/Core/Shared/Utilities/ViewHostingControl.cs]: ❕ This section of code needs comments

File: src/EditorFeatures/CSharpTest/QuickInfo/SyntacticQuickInfoSourceTests.cs [hunk 2]
```diff
@@ -293,15 +293,8 @@ private IQuickInfoProvider CreateProvider(TestWorkspace workspace)
...
 
             var viewHostingControl = (ViewHostingControl)((ElisionBufferDeferredContent)state.Content).Create();
-            try
-            {
-                var actualContent = viewHostingControl.ToString();
-                Assert.Equal(expectedContent, actualContent);
-            }
-            finally
-            {
-                viewHostingControl.TextView_TestOnly.Close();
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/QuickInfo/SyntacticQuickInfoSourceTests.cs]: ❓ Can you not close the control, and then verify that the buffer is disconnected?

Comment (jasonmalinowski) [inline src/EditorFeatures/CSharpTest/QuickInfo/SyntacticQuickInfoSourceTests.cs]: As best I can tell, this test is using ViewHostingControl as a lazy hack and probably should just test ElisionBufferDeferredContent directly.

**Comment by jasonmalinowski** [comment]
@sharwell Comments added, can you re-review? Given this is a 15.3 bug I'm trying to keep test refactoring to a minimum.

**Comment by jasonmalinowski** [comment]
Also @dotnet/roslyn-ide I need a second review for this 15.3 bug.

**Comment by jasonmalinowski** [comment]
@dotnet/roslyn-ide, please re-review this. The editor team has confirmed that we shouldn't assume anything about elision buffers, so this migrates us away from their use. This code is tricky now, so please review it with equal care.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please.

**Comment by jasonmalinowski** [comment]
@Pilchie We should probably get more testing on this and then take this to shiproom.

**Pull request merged**